cmd/otelcontribcol/go.mod

```diff
@@ -736,7 +736,7 @@ require (
 	github.com/vishvananda/netlink v1.1.1-0.20201029203352-d40f9887b852 // indirect
 	github.com/vishvananda/netns v0.0.0-20200728191858-db3c7e526aae // indirect
 	github.com/vmware/go-vmware-nsxt v0.0.0-20230223012718-d31b8a1ca05e // indirect
-	github.com/vmware/govmomi v0.36.3 // indirect
+	github.com/vmware/govmomi v0.37.3 // indirect
```

Comment:
It looks like this PR has some overlap with https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33721. Is there any chance we can wait until that one is complete so we know exactly what this PR is changing? Or if they're entirely unrelated, can we remove the `govmomi` dep upgrade changes from this PR and proceed from there?

Reply:
@crobert-1 Yeah, I heard the changes referenced in that PR were higher priority, so I'm hoping that PR gets merged in pretty quickly and then I can rebase/make this PR look a little cleaner.

Reply:
Sounds good, I'll keep an eye on it and review this once it's been rebased. Feel free to ping me if my response is delayed though 👍

Reply:
@crobert-1 should be ready to look at now!

Reply:
Thanks, it was much easier to follow after the rebase 🎉